### DISCUSSION TITLE
Device Descriptor Request failed under Windows 10

### DISCUSSION THREAD

**Discussion by joejackson3** [hathach/tinyusb#795] [Q&A]
This is my first project using the Raspberry Pico and TinyUSB, with USB HID Device functionality.
Developed with C/C++.
I have managed to get it running OK under Linux, mouse move commands are issued correctly.
When I try to connect under Windows 10, I get "USB device malfunctioned", and in Device Manager I can see that the USB device descriptor request failed. The Pico connects OK as a mass storage device in BOOTSEL mode.
In previous discussions, I noticed that someone else had a problem with failed tusb_init(). When I query this function, I too can see that this returns false.
I'm really not sure where to go from here.
Any suggestions appreciated.

**Comment by hathach** [comment]
There is a few things you could try
1. Make sure you use tinyusb from this repo instead of one included by pico-sdk
2. Run stock hid_composite example to see if that works for you.
3. Enable LOG=2 (check https://github.com/hathach/tinyusb/blob/master/docs/getting_started.md how to) and analyze the log output

**Reply by joejackson3** [reply]
Thank you for your kind suggestions. I'm very new to Github and slightly struggling to understand basics how to build my executables from your repo, which I've already cloned. If you could just throw some general keywords at me how to make use of the .mk files, I'll look it up.

**Reply by hathach** [reply]
It is in the getting started doc

**Reply by joejackson3** [reply]
Thank you. I'm trying your suggestions at the moment. It appears that in Windows 10, the handshake for Device descriptors fails (on any W10 computer). On plugging in the Pico into a Linux machine, the USB device is recognised OK and I can see the device descriptors according to the usb_descriptors.c parameters.
Could you please point me to the function in the Tinyusb stack that does the handshake between Pico and Windows 10 host, so I could debug more focussed?

**Reply by joejackson3** [reply]
I had someone help me pull the up-to-date TinyUSB repo into my project and recompile it. The problem is still exactly the same. I think the default device descriptors look correct. 
I am not sure how to go about your suggestion 2, as I don't know how to compile the stock HID_composite example.
Has anyone successfully run this on Windows 10? It works ok for me on Linux but not on Windows.
Might have to use CircuitPython in the interim - works "out of the box" with very little code.

**Reply by matefon** [reply]
I have problems on Windows 10 too. The example works fine. I've searched for possible overwrite/multiple includes that could cause the issue, couldn't find any. Tried to use the example's config files, no success.
I will try to use the example configs again, after re-pulling the repo.
